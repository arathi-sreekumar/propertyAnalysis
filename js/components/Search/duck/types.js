// @flow

export const SET_SEARCH_TERM = 'SET_SEARCH_TERM';

export default {
	SET_SEARCH_TERM
};
