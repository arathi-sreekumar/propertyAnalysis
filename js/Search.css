.search {
	background-color: white;
	padding: 20px;
	border-radius: 10px;
}

.search input {
	margin: 0 10px;
	width: auto;
}

.search .separator-text {
	font-family: Trebuchet MS;
}
