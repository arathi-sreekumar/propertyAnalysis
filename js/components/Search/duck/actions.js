// @flow

import { SET_SEARCH_TERM } from './types';

// Redux: creates an action and returns it
export function setSearchTerm(searchTerm: string) {
	return { type: SET_SEARCH_TERM, payload: searchTerm };
}

export default {
	setSearchTerm
};
