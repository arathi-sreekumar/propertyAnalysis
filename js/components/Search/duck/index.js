// @flow

import searchReducer from './reducers';

export { default as searchTypes } from './types';
export default searchReducer;
