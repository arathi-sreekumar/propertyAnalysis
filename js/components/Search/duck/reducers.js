// @flow

import { SET_SEARCH_TERM } from './types';

declare type ActionType = 'SET_SEARCH_TERM';

declare type ActionT<A: ActionType, P> = {|
	type: A,
	payload: P
|};

const searchReducer = (state: string = '', action: ActionT) => {
	switch (action.type) {
		case SET_SEARCH_TERM:
			return action.payload;
		default:
			return state;
	}
};

export default searchReducer;
