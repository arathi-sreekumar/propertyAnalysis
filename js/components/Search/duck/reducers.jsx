import { SET_SEARCH_TERM } from './types';

const DEFAULT_STATE = {
	searchTerm: ''
};

const setSearchTerm = (state, action) => Object.assign({}, state, { searchTerm: action.payload });

const searchReducer = (state = DEFAULT_STATE, action) => {
	switch (action.type) {
		case SET_SEARCH_TERM:
			return setSearchTerm(state, action);
		default:
			return state;
	}
};

export default searchReducer;
