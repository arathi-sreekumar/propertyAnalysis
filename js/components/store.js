// @flow

import { createStore, compose } from 'redux';
import rootReducer from './reducers';

// Redux store to keep all redux parameters
const store = createStore(
	rootReducer,

	/* Adding redux Devtool support */
	compose(
		typeof window === 'object' && typeof window.devToolsExtension !== 'undefined'
			? window.devToolsExtension()
			: f => f
	)
);

export default store;
