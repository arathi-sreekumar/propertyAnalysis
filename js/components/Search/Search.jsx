// @flow

import React from 'react';
import { connect } from 'react-redux';
import type { RouterHistory } from 'react-router-dom';
import { setSearchTerm } from './duck/actions';
import './Search.css';

class Search extends React.Component {
	props: {
		searchTerm: string,
		handleSearchTermChange: Function,
		history: RouterHistory
	};

	goToSearchResults = (event: SyntheticEvent) => {
		event.preventDefault();
		this.props.history.push('/searchResults');
	};

	render() {
		return (
			<form className="search field has-addons" onSubmit={this.goToSearchResults}>
				<div className="control is-expanded">
					<input
						type="text"
						className="input is-primary"
						placeholder="Search"
						value={this.props.searchTerm}
						onChange={this.props.handleSearchTermChange}
					/>
				</div>
				<div className="control">
					<button className="button is-primary" type="submit">
						Search
					</button>
				</div>
			</form>
		);
	}
}

const mapStateToProps = state => ({
	searchTerm: state.searchTerm
});

const mapDispatchToProps = (dispatch: Function) => ({
	handleSearchTermChange(event) {
		dispatch(setSearchTerm(event.target.value));
	}
});

export default connect(
	mapStateToProps,
	mapDispatchToProps
)(Search);
