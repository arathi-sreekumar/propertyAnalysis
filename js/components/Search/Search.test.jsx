import React from 'react';
import { render, mount } from 'enzyme';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import store from '../store';
import { setSearchTerm } from './duck/actions';
import Search from './Search';

jest.mock('../SearchResults/SearchResults');

describe('Search', () => {
	it('renders correctly when search word is empty', () => {
		const searchWord = '';
		store.dispatch(setSearchTerm(searchWord));
		const component = render(
			<Provider store={store}>
				<MemoryRouter>
					<Search />
				</MemoryRouter>
			</Provider>
		);
		expect(component).toMatchSnapshot();
		expect(component.find('.input').attr('value')).toBe(searchWord);
	});
	it('renders correctly for a given search word "bn10"', () => {
		const searchWord = 'bn10';
		store.dispatch(setSearchTerm(searchWord));
		const component = render(
			<Provider store={store}>
				<MemoryRouter>
					<Search />
				</MemoryRouter>
			</Provider>
		);
		expect(component).toMatchSnapshot();
		expect(component.find('.input').attr('value')).toBe(searchWord);
	});
	it('submits the form with correct url', () => {
		const history = createMemoryHistory('/');
		const spyHistoryPush = jest.spyOn(history, 'push');
		const searchWord = 'bn10';
		store.dispatch(setSearchTerm(searchWord));
		const component = mount(
			<Provider store={store}>
				<MemoryRouter>
					<Search history={history} />
				</MemoryRouter>
			</Provider>
		);
		component
			.find('form.search')
			.first()
			.simulate('submit');
		expect(spyHistoryPush).toHaveBeenCalledTimes(1);
		expect(spyHistoryPush).toHaveBeenCalledWith('/searchResults/bn10');
	});
});
