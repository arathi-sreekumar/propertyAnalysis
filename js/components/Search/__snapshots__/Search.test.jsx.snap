// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly for a given search word "bn10" 1`] = `
<form
  class="search field has-addons"
>
  <div
    class="control is-expanded"
  >
    <label
      class="screen-reader-only"
      for="search"
    >
      Search for a full or part postCode
    </label>
    <input
      class="input is-primary"
      id="search"
      placeholder="Search"
      type="text"
      value="bn10"
    />
  </div>
  <div
    class="control"
  >
    <button
      class="button is-primary"
      type="submit"
    >
      Search
    </button>
  </div>
</form>
`;

exports[`Search renders correctly when search word is empty 1`] = `
<form
  class="search field has-addons"
>
  <div
    class="control is-expanded"
  >
    <label
      class="screen-reader-only"
      for="search"
    >
      Search for a full or part postCode
    </label>
    <input
      class="input is-primary"
      id="search"
      placeholder="Search"
      type="text"
      value=""
    />
  </div>
  <div
    class="control"
  >
    <button
      class="button is-primary"
      type="submit"
    >
      Search
    </button>
  </div>
</form>
`;
